libkbfs/kbfs_ops_concur_test.go

```diff
@@ -709,6 +709,160 @@ func TestKBFSOpsConcurBlockSyncTruncate(t *testing.T) {
 	}
 }
 
+// Tests that a file that has been truncate-extended and overwritten
+// to several times can sync, and then take several deferred
+// overwrites, plus one write that blocks until the dirty bcache has
+// room.  This is a repro for KBFS-1846.
+func TestKBFSOpsTruncateAndOverwriteDeferredWithArchivedBlock(t *testing.T) {
+	config, _, ctx, cancel := kbfsOpsInitNoMocks(t, "test_user")
+	defer kbfsTestShutdownNoMocks(t, config, ctx, cancel)
+
+	bsplitter, err := NewBlockSplitterSimple(MaxBlockSizeBytesDefault, 8*1024,
+		config.Codec())
+	if err != nil {
```

Comment:
I'd rather not have a mixed use of `require` within a single test file.  It would be better to change the whole file in one PR, and then keep it up-to-date after that.